pages/p/[siteId].js

```diff
@@ -0,0 +1,84 @@
+import { useRef, useState } from 'react';
+import { useRouter } from 'next/router';
+import { Box, FormControl, FormLabel, Input, Button } from '@chakra-ui/core';
+
+import Feedback from '@/components/Feedback';
+import { useAuth } from '@/lib/auth';
+import { createFeedback } from '@/lib/db';
+import { getAllFeedback, getAllSites } from '@/lib/db-admin';
+
+export async function getStaticProps(context) {
+  const siteId = context.params.siteId;
+  const { feedback } = await getAllFeedback(siteId);
+
+  return {
+    props: {
+      initialFeedback: feedback
+    }
+  };
+}
+
+export async function getStaticPaths() {
+  const { sites } = await getAllSites();
+  const paths = sites.map((site) => ({
+    params: {
+      siteId: site.id.toString()
+    }
+  }));
+
+  return {
+    paths,
+    fallback: false
+  };
+}
+
+const FeedbackPage = ({ initialFeedback }) => {
+  const auth = useAuth();
+  const router = useRouter();
+  const inputEl = useRef(null);
+  const [allFeedback, setAllFeedback] = useState(initialFeedback);
+
+  const onSubmit = (e) => {
+    e.preventDefault();
+
+    const newFeedback = {
+      author: auth.user.name,
+      authorId: auth.user.uid,
+      siteId: router.query.siteId,
+      text: inputEl.current.value,
+      createdAt: new Date().toISOString(),
+      provider: auth.user.provider,
+      status: 'pending'
+    };
+
+    setAllFeedback([newFeedback, ...allFeedback]);
+    createFeedback(newFeedback);
+  };
+
+  return (
+    <Box
+      display="flex"
+      flexDirection="column"
+      width="full"
+      maxWidth="700px"
+      margin="0 auto"
+    >
+      {auth.user && (
+        <Box as="form" onSubmit={onSubmit}>
```

Comment:
When working with forms in React, when would you decide to use `react-hook-form`?

Reply:
I generally switch over to `react-hook-form` if I need input validation. I might add that in later to ensure certain standards for leaving feedback.